app/src/main/java/fr/free/nrw/commons/upload/SingleUploadFragment.java

```diff
@@ -104,13 +105,75 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         Log.d("Single Upload fragment", license);
 
-        ArrayAdapter<String> adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_dropdown_item, licenseItems);
+        ArrayAdapter<String> adapter;
+        if (PreferenceManager.getDefaultSharedPreferences(getActivity()).getBoolean("theme",true)) {
+            // dark theme
+            adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_dropdown_item, licenseItems);
+        }else {
+            // light theme
+            adapter = new ArrayAdapter<>(getActivity(), R.layout.light_simple_spinner_dropdown_item, licenseItems);
+        }
+
         licenseSpinner.setAdapter(adapter);
 
         int position = licenseItems.indexOf(getString(Utils.licenseNameFor(license)));
         Log.d("Single Upload fragment", "Position:"+position+" "+getString(Utils.licenseNameFor(license)));
         licenseSpinner.setSelection(position);
 
+        licenseSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+
+                //Set selected color to white because it should be readable on random images.
+                TextView selectedText = (TextView) licenseSpinner.getChildAt(0);
+                if (selectedText != null ) {
+                    selectedText.setTextColor(Color.WHITE);
+                    selectedText.setBackgroundColor(Color.TRANSPARENT);
+                }
+
+                String licenseName = parent.getItemAtPosition(position).toString();
+
+                String license = Prefs.Licenses.CC_BY_SA_3; // default value
+                if(getString(R.string.license_name_cc0).equals(licenseName)) {
+                    license = Prefs.Licenses.CC0;
+                } else if(getString(R.string.license_name_cc_by).equals(licenseName)) {
+                    license = Prefs.Licenses.CC_BY_3;
+                } else if(getString(R.string.license_name_cc_by_sa).equals(licenseName)) {
+                    license = Prefs.Licenses.CC_BY_SA_3;
+                } else if(getString(R.string.license_name_cc_by_four).equals(licenseName)) {
+                    license = Prefs.Licenses.CC_BY_4;
+                } else if(getString(R.string.license_name_cc_by_sa_four).equals(licenseName)) {
+                    license = Prefs.Licenses.CC_BY_SA_4;
+                }
+
+                setLicenseSummary(license);
+                SharedPreferences.Editor editor = prefs.edit();
+                editor.putString(Prefs.DEFAULT_LICENSE, license);
+                editor.apply();
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        titleDescButton.setOnClickListener(new View.OnClickListener()
```

Comment:
This is duplicated. On click listener is already defined below using butterknife. I guess it got duplicated while rebasing.